tests/openvino/test_quantization.py

```diff
@@ -162,6 +195,176 @@ def test_aware_training_quantization(self, model_name, expected_fake_quantize, e
+        expected_fake_quantize = 19
+        expected_int8 = 14
+
+        ov_config = OVConfig(compression=[MOVEMENT_SPARSITY_CONFIG_FOR_BERT, DEFAULT_QUANTIZATION_CONFIG])
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+        model = AutoModelForSequenceClassification.from_pretrained(model_name)
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            ov_config.log_dir = tmp_dir
+            trainer = self.build_glue_sst_trainer(
+                tmp_dir,
+                tokenizer=tokenizer,
+                model=model,
+                ov_config=ov_config,
+            )
+            trainer.train()
+            trainer.save_model()
+            self.assertIn("compression_loss", trainer.compression_metrics)
+
+            state_dict = torch.load(Path(tmp_dir, WEIGHTS_NAME), map_location="cpu")
+            num_binary_masks = sum(key.endswith("_binary_mask") for key in state_dict)
+            self.assertEqual(expected_binary_masks, num_binary_masks)
+
+            ovmodel = OVModelForSequenceClassification.from_pretrained(tmp_dir)
+            num_fake_quantize, num_int8 = self.count_quantization_op_number(ovmodel)
+            self.assertEqual(expected_fake_quantize, num_fake_quantize)
+            self.assertEqual(expected_int8, num_int8)
+
+            tokens = generate_mock_tokens(trainer.model.input_infos)
+            outputs = ovmodel(**tokens)
+            self.assertTrue("logits" in outputs)
+
+    def test_training_movement_sparsity_quantization_distillation(self):
+        model_name = "hf-internal-testing/tiny-bert"
+        teacher_model_name = "hf-internal-testing/tiny-bert"
+        expected_binary_masks = 24
+        expected_fake_quantize = 19
+        expected_int8 = 14
+
+        ov_config = OVConfig(compression=[MOVEMENT_SPARSITY_CONFIG_FOR_BERT, DEFAULT_QUANTIZATION_CONFIG])
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+        model = AutoModelForSequenceClassification.from_pretrained(model_name)
+        teacher_model = AutoModelForSequenceClassification.from_pretrained(teacher_model_name)
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            ov_config.log_dir = tmp_dir
+            trainer = self.build_glue_sst_trainer(
+                tmp_dir,
+                tokenizer=tokenizer,
+                model=model,
+                teacher_model=teacher_model,
+                ov_config=ov_config,
+            )
+            trainer.train()
+            trainer.save_model()
+            self.assertIn("distillation_loss", trainer.compression_metrics)
+            self.assertIn("compression_loss", trainer.compression_metrics)
+
+            state_dict = torch.load(Path(tmp_dir, WEIGHTS_NAME), map_location="cpu")
+            num_binary_masks = sum(key.endswith("_binary_mask") for key in state_dict)
+            self.assertEqual(expected_binary_masks, num_binary_masks)
+
+            ovmodel = OVModelForSequenceClassification.from_pretrained(tmp_dir)
```

Comment:
It could be interesting to also compare the outputs between the `trainer.model` and the `ovmodel` to verify the loading, as done [here](https://github.com/huggingface/optimum-intel/blob/main/tests/neural_compressor/test_optimization.py#L256) (I can do it in a following PR as well)

Reply:
@vuiseng9, please address this

Reply:
@AlexKoff88 the torch model here could have a different input shape to the IR model. We faced this issue and temporarily using mock tokens to check if IR model can be inferred. Once the dynamic shape IR issue is resolved, we can compare the `trainer.model` and `ovmodel` output.

Reply:
@echarlaix @AlexKoff88 since we have dynamic IR now, we also cover this test now. See https://github.com/huggingface/optimum-intel/pull/150/commits/6a93a79325656fcc3307b3d8759630238d058b87 .